components/shared/compositing/constellation_msg.rs

```diff
@@ -102,15 +104,16 @@ impl fmt::Debug for ConstellationMsg {
             LogEntry(..) => "LogEntry",
             NewBrowser(..) => "NewBrowser",
             CloseBrowser(..) => "CloseBrowser",
+            FocusBrowser(..) => "FocusBrowser",
+            UnfocusBrowser => "UnfocusBrowser",
             SendError(..) => "SendError",
-            SelectBrowser(..) => "SelectBrowser",
             ForwardEvent(..) => "ForwardEvent",
             SetCursor(..) => "SetCursor",
             EnableProfiler(..) => "EnableProfiler",
             DisableProfiler => "DisableProfiler",
             ExitFullScreen(..) => "ExitFullScreen",
             MediaSessionAction(..) => "MediaSessionAction",
-            ChangeBrowserVisibility(..) => "ChangeBrowserVisibility",
+            WindowVisibility(..) => "WindowVisibility",
```

Comment:
I think this is a bit funky. The embedder should have a concept of a window, but I'm not sure that Constellation should (at least for now). Instead, the embedder should send a visibility event for each web view when their window is hidden. The reason I think this is that maybe the embedder wants to move a web view to a different window. This should totally be possible without destroying it and recreating the web view completely. If the Constellation is managing associating web views to windows, then it will require complex bookkeeping in both the embedder and the Constellation to track movement of web views to different windows.

Reply:
i agree. fixed in 9e307cfcefed26ecb65399ce12a7e2a7ef38e5d7